packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts

```diff
@@ -0,0 +1,22 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import * as plugins from 'temp/middleware-plugins';
+
+export interface MiddlewarePlugin {
+  /**
+   * Detect order when the plugin should be called, e.g. 0 - will be called first (can be a plugin which data is required for other plugins)
+   */
+  order: number;
+  /**
+   * A middleware to be called, it's required to return @type {NextResponse} for other middlewares
+   */
+  (req: NextRequest, res: NextResponse): Promise<NextResponse>;
+}
+
+export default async function middleware(req: NextRequest): Promise<NextResponse> {
```

Comment:
Should we add the `NextFetchEvent` parameter and also pass along to each plugin?

Reply:
Done